core/domain/learner_goals_services_test.py

```diff
@@ -0,0 +1,252 @@
+        topic_services.publish_topic(self.TOPIC_ID_4, self.admin_id)
+
+    def _get_all_topic_ids_to_learn(self, user_id):
+        """Returns the list of all the topic ids to learn
+        corresponding to the given user id.
+        """
+        learner_goals_model = user_models.LearnerGoalsModel.get(
+            user_id, strict=False)
+
+        return (
+            learner_goals_model.topic_ids_to_learn if
+            learner_goals_model else [])
+
+    def test_single_topic_is_added_correctly_to_learn(self):
+        # Test adding a single topic_id to learn.
+        self.assertEqual(
+            self._get_all_topic_ids_to_learn(self.viewer_id), [])
+        learner_progress_services.validate_and_add_topic_to_learn_goal(
+            self.viewer_id, self.TOPIC_ID_1)
+        self.assertEqual(
+            self._get_all_topic_ids_to_learn(
+                self.viewer_id), [self.TOPIC_ID_1])
+
+    def test_multiple_topics_are_added_correctly_to_learn(self):
+        # Test adding two topics to the learn.
+        self.assertEqual(
+            self._get_all_topic_ids_to_learn(
+                self.viewer_id), [])
+
+        learner_progress_services.validate_and_add_topic_to_learn_goal(
+            self.viewer_id, self.TOPIC_ID_1)
+        self.assertEqual(
+            self._get_all_topic_ids_to_learn(
+                self.viewer_id), [self.TOPIC_ID_1])
+
+        learner_progress_services.validate_and_add_topic_to_learn_goal(
+            self.viewer_id, self.TOPIC_ID_2)
+        self.assertEqual(
+            self._get_all_topic_ids_to_learn(
+                self.viewer_id), [self.TOPIC_ID_1, self.TOPIC_ID_2])
+
+    def test_adding_exisiting_topic_is_not_added_again(self):
+        # Test adding the topic_id if it is already in
+        # learner_goals.topic_id.
+        learner_progress_services.validate_and_add_topic_to_learn_goal(
+            self.viewer_id, self.TOPIC_ID_1)
+        learner_progress_services.validate_and_add_topic_to_learn_goal(
+            self.viewer_id, self.TOPIC_ID_2)
+        self.assertEqual(
+            self._get_all_topic_ids_to_learn(
+                self.viewer_id), [self.TOPIC_ID_1, self.TOPIC_ID_2])
+
+        learner_progress_services.validate_and_add_topic_to_learn_goal(
+            self.viewer_id, self.TOPIC_ID_1)
+        self.assertEqual(
+            self._get_all_topic_ids_to_learn(
+                self.viewer_id), [self.TOPIC_ID_1, self.TOPIC_ID_2])
+
+    def test_completed_topic_is_not_added_to_learner_goals(self):
+        learner_progress_services.validate_and_add_topic_to_learn_goal(
+            self.viewer_id, self.TOPIC_ID_1)
+        self.assertEqual(
+            self._get_all_topic_ids_to_learn(
+                self.viewer_id), [self.TOPIC_ID_1])
+
+        learner_progress_services.mark_topic_as_learnt(
+            self.viewer_id, self.TOPIC_ID_2)
+
+        # Test that the topic added to the in learnt list doesn't get
```

Comment:
'the learnt list'. Shouldn't there be one more check where if a topic is learnt, it is removed from the goals? Or is that tested elsewhere?

Reply:
Done. And it is tested in [learner_progress_services_test](https://github.com/oppia/oppia/pull/13081/files#diff-2a96ae1edea930d882ddc60b43d581c34db5e827bd169f51a30de3caccbbacefR568). PTAL!